Penteract/Assets/Scripts/PauseController.cpp

```diff
@@ -0,0 +1,63 @@
+#include "PauseController.h"
+
+#include "GameObject.h"
+#include "GameplaySystems.h"
+
+EXPOSE_MEMBERS(PauseController) {
+	// Add members here to expose them to the engine. Example:
+	// MEMBER(MemberType::BOOL, exampleMember1),
+	// MEMBER(MemberType::PREFAB_RESOURCE_UID, exampleMember2),
+	// MEMBER(MemberType::GAME_OBJECT_UID, exampleMember3)
+};
+
+GENERATE_BODY_IMPL(PauseController);
+
+void PauseController::Start() {
+
+	//Get audio sources
+	int i = 0;
+
+	for (ComponentAudioSource& src : GetOwner().GetComponents<ComponentAudioSource>()) {
+		if (i < static_cast<int>(AudioPause::TOTAL)) audios[i] = &src;
+		i++;
+	}
+
+	levelMusic = GetOwner().GetParent()->GetComponent<ComponentAudioSource>();
+}
+
+void PauseController::Update() {
+	if (!hasEntered && isPause) {
+		hasEntered = true;
+		if (audios[static_cast<int>(AudioPause::OPEN)]) {
+			audios[static_cast<int>(AudioPause::OPEN)]->Play();
+		}
```

Comment:
Since you are using the same 3 lines in the isPause and !isPause statement, you can create a method that takes the Enum AudioPause as input so you don't have the 3 same lines repeated in 4 different places. Also, it would be appropiate to check the size of audios before accessing it to avoid OutOfBounds Exceptions (this would only occur if the GameObject is changed in the hierarchy)

Reply:
As we discussed, we will keep this version